### ISSUE TITLE
Training data is reused for evaluation of the model

### ISSUE THREAD

**Issue by minaoar** [tamanna037/InformationTypesDetectionNLP#3]
**Current Scenario**
In the [source code](https://github.com/tamanna037/InformationTypesDetectionNLP/blob/main/code/InfoTypeDetectionOSSProject-Tamanna.ipynb), the classification model is trained using a training and a validation data set. However, while evaluating the trained model, instead of using a separate test data set, previously used validation data set (df_val) is provided. Here is the screenshot of the source code:

> ![image](https://user-images.githubusercontent.com/5583535/195506350-aa9eea55-3086-4792-ae98-1b172c5217d2.png)

**Impact**
When training data is provided for model performance evaluation, the accuracy of model is wrongly reported higher because the model is already familiar with this training data. You can find related online references [here](https://machinelearningmastery.com/a-simple-intuition-for-overfitting/). Here is an excerpt from another [blog](https://towardsdatascience.com/rookie-errors-in-machine-learning-bc1c627f2789):

> Mixing training data into your validation or test set is easy to do, and will produce superb model performance — as assessed by your dodgy test set — and terrible performance in the real world.

**Resolution**
The code already has test data set in df_test which is not used in training. In the validation method (eval_model), this df_test can be passed. Here is the proposed code:

> result, model_outputs, wrong_predictions = model.eval_model(**df_test**)
> print(result)

**Further data as requested in Contributing file**
*Environment:* this error (wrong high accuracy of the model) will be reproduced irrespective of any environment.
*Steps to Reproduce:* just running the code up to validation part will reproduce the error. The difference of model accuracy (result) can be found after evaluating the model with test data set.
*Expected result:* how the model performs which unknown new data set
*Actual result:* how the model performs against the data set it has already seen. The accuracy will be reported higher than expected.
*Severity:* this does not impact the prediction of new/unknown datasets. However, it misrepresents the actual prediction accuracy  of the trained model.

**Comment by tamanna037** [comment]
Thank you for pointing out the bug. It was a typo. You can pull request to fix the bug as you have suggested. I think that can be a good way for contributing this project.

**Comment by minaoar** [comment]
If you accept the change, could you please add me as a contributor? My ID is minaoar.

**Comment by tamanna-sb** [comment]
The eval_model() method is used to evaluate the model, not predicting. For predicting, The predict() method  will be used and test data will be passed as parameter here. 

SInce in the training, already evaluation on the validation data is done, using eval_model was repetitive, not wrong. It will provide the same results what we have got earlier in the training procedure. I am not measuring the model efficiency based on the result of eval_model method. Result of  predict(test_df) method will be the real testing.

Here is a [link](https://simpletransformers.ai/docs/classification-models/) which describes what each method does.

**Comment by minaoar** [comment]
That's interesting. Thanks for sharing the details. Somehow the link is not updated in your above comment.

**Comment by tamanna-sb** [comment]
I have updated the link. Also sharing here, https://simpletransformers.ai/docs/classification-models/
